torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1722,34 +1722,36 @@ def rref_to_here(rref_var):
     @dist_init
     def test_remote_script_module(self):
         @torch.jit.ignore
-        def my_script_module_init():
-            # type: () -> MyModuleInterface
-            return MyScriptModule()
+        def my_script_module_init(rank):
+            # type: (int) -> MyModuleInterface
+            return MyScriptModule(rank)
 
         @torch.jit.script
-        def construct_my_script_module():
-            # type: () -> MyModuleInterface
-            return my_script_module_init()
-
-        n = self.rank + 1
-        dst_rank = n % self.world_size
-        ref_script_module = rpc.remote(
-            "worker{}".format(self.rank),
-            construct_my_script_module,
-            args=())
+        def construct_my_script_module(rank):
+            # type: (int) -> MyModuleInterface
+            return my_script_module_init(rank)
 
         @torch.jit.script
         def run_ref_script_module(ref_script_module):
             # type: (RRef[MyModuleInterface]) -> Tensor
             module = ref_script_module.to_here()
             return module.forward()
 
-        local_ret = run_ref_script_module(ref_script_module)
-        print(local_ret)
-        '''
+        # TODO, need more investigation
+        # there is rref leak when shutting down, suspect it is because
+        # ref as arg is passed to pybind boundary, and the ref is not garbage
+        # collected by python when calling shutdown()
+        import torch.distributed.rpc.api as api
+        api._ignore_rref_leak = True
```

Comment:
@mrshenli How do other RRef tests guarantee that RRefs are garbage collected before shutdown() is called?

Reply:
Other tests rely on Python to deref local RRef objects when exiting the test function context, and then uses agent.join() to wait for all messages settle.